src/lib/mina/v1/transaction-validation.ts

```diff
@@ -331,6 +344,13 @@ type AuthorizationKind = { isProved: boolean; isSigned: boolean };
 
 const isPair = (a: AuthorizationKind, b: AuthorizationKind) => !a.isProved && !b.isProved;
 
+const isValidVk = (vk: VerificationKey) => {
+  const verificationKey = VerificationKey.fromValue(vk);
+  const circuitVk = Pickles.sideLoaded.vkToCircuit(() => verificationKey.data);
+  const inCircuitHash = inCircuitVkHash(circuitVk);
```

Comment:
nit: would be cleaner to export an out-of-circuit hashing method from ocaml for this purpose

Reply:
agree, would be nice to do this